File: cpp/Kode program/ApplianceItem.cpp
#pragma once
#include <iostream>
#include "HouseOldItem.cpp"
#include "BrandInfo.cpp"
#include "PowerTypeInfo.cpp"
#include "WarrantyInfo.cpp"
using namespace std;

class ApplianceItem : public HouseOldItem, public BrandInfo {
private:
    string powerType;
    string warranty;
    PowerTypeInfo powerInfo;    // Komposisi dengan PowerTypeInfo
    WarrantyInfo warrantyInfo;  // Komposisi dengan WarrantyInfo
public:
    void setPowerType(string pt) { powerType = pt; }  // Set tipe daya
    string getPowerType() { return powerType; }       // Get tipe daya

    void setWarranty(string w) { warranty = w; }  // Set garansi
    string getWarranty() { return warranty; }     // Get garansi

    void setPowerTypeInfo(PowerTypeInfo pi) { powerInfo = pi; }  // Set info daya
    PowerTypeInfo getPowerTypeInfo() { return powerInfo; }       // Get info daya

    void setWarrantyInfo(WarrantyInfo wi) { warrantyInfo = wi; }  // Set info garansi
    WarrantyInfo getWarrantyInfo() { return warrantyInfo; }       // Get info garansi

    void printAppliance() {  // Print info appliance lengkap
        cout << "=== Appliance Item ===" << endl;
        printItem();      // Dari HouseOldItem
        printBrand();     // Dari BrandInfo
        cout << "Power Type: " << powerType << endl;
        cout << "Warranty: " << warranty << endl;
    }
};

File: python/Kode program/ApplianceItem.py
from HouseOldItem import HouseOldItem
from BrandInfo import BrandInfo
from PowerTypeInfo import PowerTypeInfo
from WarrantyInfo import WarrantyInfo

class ApplianceItem(HouseOldItem, BrandInfo):
    def __init__(self):
        HouseOldItem.__init__(self)
        BrandInfo.__init__(self)
        self.powerType = ""
        self.warranty = ""
        self.powerInfo = PowerTypeInfo()
        self.warrantyInfo = WarrantyInfo()

    def setPowerType(self, pt):
        self.powerType = pt

    def getPowerType(self):
        return self.powerType

    def setWarranty(self, w):
        self.warranty = w

    def getWarranty(self):
        return self.warranty

    def setPowerTypeInfo(self, pi):
        self.powerInfo = pi

    def getPowerTypeInfo(self):
        return self.powerInfo

    def setWarrantyInfo(self, wi):
        self.warrantyInfo = wi

    def getWarrantyInfo(self):
        return self.warrantyInfo

    def printAppliance(self):
        print("=== Appliance Item ===")
        self.printItem()
        self.printBrand()
        print(f"Power Type: {self.powerType}")
        print(f"Warranty: {self.warranty}")
